statistics/scalar.go

```diff
@@ -175,3 +177,122 @@ func calcFraction4Datums(lower, upper, value *types.Datum) float64 {
 	}
 	return 0.5
 }
+
+const maxNumStep = 10
+
+func enumRangeValues(low, high types.Datum, lowExclude, highExclude bool) []types.Datum {
+	if low.Kind() != high.Kind() {
+		return nil
+	}
+	exclude := 0
+	if lowExclude {
+		exclude++
+	}
+	if highExclude {
+		exclude++
+	}
+	switch low.Kind() {
+	case types.KindInt64:
+		// Overflow check.
+		lowVal, highVal := low.GetInt64(), high.GetInt64()
+		if lowVal < 0 && highVal > 0 {
+			if lowVal <= -maxNumStep || highVal >= maxNumStep {
```

Comment:
Why do we need to limit the range in `[-maxNumStep, maxNumStep]`?

Reply:
Overflow check, combined with the upper `if`, we can make sure that it is safe and does not miss valid ranges.

Reply:
I think these two `if` are preventing overflow `int64`.

Reply:
got